engine/sql/core/src/main/scala/org/apache/spark/sql/index/TreeMapIndexedRelation.scala

```diff
@@ -50,10 +49,9 @@ private[sql] case class TreeMapIndexedRelation(
       val eval_key = BindReferences.bindReference(column_keys.head, child.output).eval(row)
       eval_key match {
         case key: Number => (key.doubleValue, row)
-        case key: GenericInternalRow =>
-          val row_array = key.values(0).asInstanceOf[GenericArrayData].array
-          require(row_array.length == 1)
-          (row_array.head.asInstanceOf[Double], row)
+        case key: Point =>
```

Comment:
We should not support TreeMap Index for Point. There is no way to use a one-dimensional index for multi-dimensional data.